src/external/serde_support.rs

```diff
@@ -138,6 +129,24 @@ impl<'de> Deserialize<'de> for Uuid {
     }
 }
 
+macro_rules! deserialize_uuid_proxy {
+    ($type:ty) => {
+        impl<'de> Deserialize<'de> for $type {
+            fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+            where
+                D: serde::Deserializer<'de>,
+            {
+                Ok(Uuid::deserialize(deserializer)?.into())
```

Comment:
If you're trying to deserialize a UUID into one of these proxy types, I think we should only accept a UUID in its format. We should be able to make the visitor used in `Uuid`'s deserialize impl generic and add appropriate `FromStr` implementations to the proxy types. Then the bound on that visitor would be any `T: FromStr + From<Uuid>`.

I'd be happy to sketch it out a bit further if that would be helpful 🙂

Reply:
Alright, I can change it to have that behavior